src/dmd/ctfeexpr.d

```diff
@@ -746,6 +746,17 @@ Expression getAggregateFromPointer(Expression e, dinteger_t* ofs)
             return se.e1;
         }
     }
+    if (auto ie = e.isIntegerExp())
+        if (ie.type.ty == Tpointer)
+        {
+            // It can be a `null` disguised as a cast, e.g. `cast(void*)0`.
+            if (ie.getInteger() == 0)
```

Comment:
Is there any reason not to conflate this `if` with the one above by using `&&` ?

Reply:
Done